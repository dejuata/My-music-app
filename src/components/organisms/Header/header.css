.header {
    display: flex;
    align-items: center;
    justify-content: space-around;
    height: 3em;
    width: 100%;
    position: sticky;
    top: 0;
    color: var(--color-dark);
    background: var( --color-white);
    font-size: var(--sm);
    z-index: 100;
}

.header p {
    margin: 0;
    font-size: 12px;
}

.header__logo {
    display: flex;
    align-items: center;
    column-gap: 6px;
    font-size: var(--sm);
    font-weight: bold;
}

.header__logo img {
    width: 34px;
    height: 34px;
}