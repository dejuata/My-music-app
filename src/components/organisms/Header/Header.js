import React from 'react'

import './header.css';

export const Header = () => {

    const logoPath = `/assets/logo.svg`;

    return (
        <header className='header'>
            <div className='header__logo'>
                <img src={logoPath} alt="logo"/>
                MY MUSIC APP
            </div>
            <p>Juan David Pino</p>
        </header>
    )
}
