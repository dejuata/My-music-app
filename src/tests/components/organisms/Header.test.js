import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';

import { mount } from 'enzyme';
import { Provider } from "react-redux";
import { Header } from "../../../components/organisms/Header/Header";

const middlewares = [ thunk ]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {
        name: 'Juan David'
    },
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);

describe('test Header', () => { 

    const wrapper = mount(
        <Provider store={ store }>
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        </Provider>
    )

    test('should be render Header', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('should be display name user', () => {

        expect( wrapper.find('p').prop('children') ).toBe( initState.auth.name )

    })
})