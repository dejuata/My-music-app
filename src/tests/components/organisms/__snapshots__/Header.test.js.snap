// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header should be render Header 1`] = `
<header
  className="o-header"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <div
      className="o-header__logo"
    >
      <img
        alt="logo"
        src="/assets/logo.svg"
      />
      MY MUSIC APP
    </div>
  </a>
  <p>
    Juan David
  </p>
</header>
`;
